dojson/overdo.py

```diff
@@ -55,7 +55,9 @@ def do(self, blob):
 
         for key, value in iteritems(blob):
             for name, creator, field in self._query(key):
-                output[name] = creator(output, key, value)
+                val = creator(output, key, value)
```

Comment:
Nice approach. Do you think we have to do this in the same decorator or we have to split into two functions?